.github/workflows/foundry-reports.yml

```diff
@@ -0,0 +1,89 @@
+name: Foundry Reports
+
+on: [push]
+
+concurrency:
+   group: ${{github.workflow}}-${{github.ref}}
+   cancel-in-progress: true
+
+env:
+  MAINNET_RPC: ${{ secrets.MAINNET_RPC }}
+  SEPOLIA_RPC: ${{ secrets.SEPOLIA_RPC }}
+
+jobs:
+  generate-gas-report:
+    name: Generate Foundry Gas Reports
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+          cache: 'yarn'
```

Comment:
Could you upgrade those actions to v4? See an example here: https://github.com/defi-wonderland/solidity-foundry-boilerplate/blob/main/.github/workflows/release.yml#L18-L31